connectors/kafka/src/main/java/forklift/controller/KafkaController.java

```diff
@@ -145,32 +114,20 @@ public void start() {
      * @throws InterruptedException if the thread is interrupted
      */
     public void stop(long timeout, TimeUnit timeUnit) throws InterruptedException {
-        kafkaConsumer.wakeup();
-        executor.shutdownNow();
-
         running = false;
+        kafkaConsumer.wakeup();
+       // executor.shutdownNow();
+        executor.shutdown();
```

Comment:
I think you meant shutdownNow causes an interrupt?  I originally thought that I wouldn't need either of the shutdown calls as well when I did this refactor, but discovered that the awaitTermination requires a shutdown call of some sort to function.   I'll remove the commented shutdownNow